detekt-report-sarif/src/main/kotlin/io/github/detekt/report/sarif/Results.kt

```diff
@@ -43,3 +48,8 @@ private fun Issue.Location.toLocation(): io.github.detekt.sarif4k.Location =
             )
         )
     )
+
+private fun String.sha1(): String = MessageDigest
+    .getInstance("SHA-1")
+    .digest(toByteArray())
+    .joinToString(separator = "", transform = { "%02x".format(Locale.ROOT, it) })
```

Comment:
```suggestion
    .joinToString("") { it.toUByte().toString(16).padStart(2, '0') }
```

should also work, not sure which one we prefer in terms of readability

Reply:
I like yours more